src/CityOnPlanet.cpp

```diff
@@ -16,52 +17,56 @@
 #include "scenegraph/ModelSkin.h"
 #include "scenegraph/SceneGraph.h"
 
-static const unsigned int DEFAULT_NUM_BUILDINGS = 1000;
-static const double START_SEG_SIZE = CityOnPlanet::RADIUS;
-static const double START_SEG_SIZE_NO_ATMO = CityOnPlanet::RADIUS / 5.0f;
+#define DEBUG_OUTPUT 1
+
+static const char *CITY_CONFIG_FILE = "models/buildings/details.json";
+static uint8_t citydistancegrid[CityOnPlanet::citygridlimit][CityOnPlanet::citygridlimit];
+static double population2radius[5];
+static double cityradius_mod[5];
+static uint32_t cityradius_rnd[5];
+
+int simplebuildings[50];
+int numberofSimplebuildings;
+
+static const uint32_t DEFAULT_ATMO_SIZE = 60;
+static const uint32_t DEFAULT_AIRLESS_SIZE = 20;
+uint32_t config_atmo_size;
+uint32_t config_airless_size;
 
 using SceneGraph::Model;
 
 CityOnPlanet::citybuildinglist_t CityOnPlanet::s_buildingList = {
 	"city_building",
-	800,
-	2000,
 	0,
 	0,
 };
 
-CityOnPlanet::cityflavourdef_t CityOnPlanet::cityflavour[CITYFLAVOURS];
-
 void CityOnPlanet::AddStaticGeomsToCollisionSpace()
 {
 	// reset data structures
 	m_enabledBuildings.clear();
 	m_buildingCounts.resize(s_buildingList.numBuildings);
-	for (Uint32 i = 0; i < s_buildingList.numBuildings; i++) {
+	for (uint32_t i = 0; i < s_buildingList.numBuildings; i++) {
 		m_buildingCounts[i] = 0;
 	}
 
 	// Generate the new building list
 	int skipMask;
-	switch (Pi::detail.cities) {
-	case 0: skipMask = 0xf; break;
-	case 1: skipMask = 0x7; break;
-	case 2: skipMask = 0x3; break;
-	case 3: skipMask = 0x1; break;
-	default:
+	if (Pi::detail.cities >= 0 && Pi::detail.cities <= 3)
+		skipMask = (16 >> Pi::detail.cities) - 1;
+	else
```

Comment:
I would appreciate it if you would add a comment here explaining the higher-level purpose of the skip mask for easier reading of this function.